### PR TITLE
Use the `Address` class from sdk-py-core

### PR DESCRIPTION
**Pull request by popenta**

### CODE REVIEW

File: multiversx_sdk_cli/transactions.py [hunk 1]
```diff
@@ -196,8 +199,8 @@ def to_dictionary(self) -> Dict[str, Any]:
     # Creates the payload for a "user" / "inner" transaction
     def to_dictionary_as_inner(self) -> Dict[str, Any]:
         dictionary = self.to_dictionary()
-        dictionary["receiver"] = base64.b64encode(Address(self.receiver).pubkey()).decode()
-        dictionary["sender"] = base64.b64encode(Address(self.sender).pubkey()).decode()
+        dictionary["receiver"] = base64.b64encode(bytes.fromhex(Address.from_bech32(self.receiver).hex())).decode()
+        dictionary["sender"] = base64.b64encode(bytes.fromhex(Address.from_bech32(self.sender).hex())).decode()
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk_cli/transactions.py:203]: Yes, now using `Address.from_bech32().pubkey`

File: multiversx_sdk_cli/guards.py [hunk 2]
```diff
@@ -12,14 +13,14 @@ def is_directory(directory: Path):
         raise errors.BadDirectory(str(directory))
 
 
-def is_hex_address(input):
+def is_hex_address(input: str):
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk_cli/guards.py:16]: Deleted

File: multiversx_sdk_cli/guards.py [hunk 3]
```diff
@@ -12,14 +13,14 @@ def is_directory(directory: Path):
...
     is_hex_string(input)
 
     if len(input) != 64:
         raise errors.BadInputError(input, "is not a valid hex-encoded address")
 
 
-def is_hex_string(input):
+def is_hex_string(input: str):
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk_cli/guards.py:23]: Deleted

File: multiversx_sdk_cli/contracts.py [hunk 4]
```diff
@@ -104,12 +105,12 @@ def compute_address(self):
         """
         8 bytes of zero + 2 bytes for VM type + 20 bytes of hash(owner) + 2 bytes of shard(owner)
         """
-        owner_bytes = self.owner.address.pubkey()
+        owner_bytes = bytes.fromhex(self.owner.address.hex())
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk_cli/contracts.py:108]: Removed `compute_address()`, now using `compute_contract_address()` from sdk-core

File: multiversx_sdk_cli/cli_network.py [hunk 5]
```diff
@@ -27,15 +30,15 @@ def setup_parser(subparsers: Any) -> Any:
     return subparsers
 
 
-def get_num_shards(args: Any):
+def get_num_shards(args: Any) -> int:
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk_cli/cli_network.py:33]: i am not sure why these should be replaced but here's the [issue](https://github.com/multiversx/mx-sdk-py-cli/issues/285).

File: multiversx_sdk_cli/cli_wallet.py [hunk 6]
```diff
@@ -234,16 +236,21 @@ def _create_wallet_content(
...
+        result = address.hex()
+
     print(result)
     return result
 
 
 def pem_address(args: Any):
     account = Account(pem_file=args.pem, pem_index=args.pem_index)
-    print(account.address)
+    print(account.address.bech32())
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk_cli/cli_wallet.py:253]: I agree. I don't really see it's use case.

File: multiversx_sdk_cli/tests/test_accounts.py [hunk 7]
```diff
@@ -2,36 +2,32 @@
...
+from multiversx_sdk_cli.accounts import Account
 from multiversx_sdk_cli.transactions import Transaction
+from multiversx_sdk_cli.constants import DEFAULT_HRP
+from multiversx_sdk_core import Address
 
 
 def test_address():
-    address = Address("erd1l453hd0gt5gzdp7czpuall8ggt2dcv5zwmfdf3sd3lguxseux2fsmsgldz")
-    address_cloned = Address(address)
+    address = Address.from_bech32("erd1l453hd0gt5gzdp7czpuall8ggt2dcv5zwmfdf3sd3lguxseux2fsmsgldz")
```
[the remark this answers is not in the document]
Comment (popenta) [reply multiversx_sdk_cli/tests/test_accounts.py:12]: Deleted!

**Review by andreibancioiu — APPROVED** [review]

File: multiversx_sdk_cli/constants.py [hunk 8]
```diff
@@ -6,3 +6,5 @@
 DEFAULT_CARGO_TARGET_DIR_NAME = "default_cargo_target"
 TRANSACTION_OPTIONS_TX_GUARDED = 0b0010
 ONE_YEAR_IN_SECONDS = 60 * 60 * 24 * 365
+DEFAULT_HRP = "erd"
+ADDRESS_ZERO = "erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu"
```
Comment (bogdan-rosianu) [inline multiversx_sdk_cli/constants.py:10]: might rename it to `ADDRESS_ZERO_BECH32` ?

File: multiversx_sdk_cli/tests/test_proxy.py [hunk 9]
```diff
@@ -55,7 +56,8 @@ def test_get_hyperblock_by_hash():
 
 
 def test_sync_nonce():
-    account = Account("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th")
+    account = Account()
```
Comment (bogdan-rosianu) [inline multiversx_sdk_cli/tests/test_proxy.py:59]: why not `account = Account(address = Address.from_bech32("erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th"))` ?

File: multiversx_sdk_cli/constants.py (hunk 8 above)
Comment (popenta) [reply multiversx_sdk_cli/constants.py:10]: Renamed

File: multiversx_sdk_cli/tests/test_proxy.py (hunk 9 above)
Comment (popenta) [reply multiversx_sdk_cli/tests/test_proxy.py:59]: Done! Also the test is a bit "fragile" and will soon fail after the reset of the devnet.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**